### ISSUE TITLE
User interface

### ISSUE THREAD

**Issue by cvvergara** [pgRouting/osm2pgrouting#43]
Program parameters, like name of file, database.
Basically from https://github.com/pgRouting/osm2pgrouting/blob/master/src/osm2pgrouting.cpp#L37
to L141 is to be “transformed” to use boost::program_options

Difficulty: Easy
Urgency: First things first. This is first
I would like Sarthak to define the help message for each option. I think it will help him understand why we require what we require. (some things are obvious, but what is obvious for me its not obvious for all), If he has questions then in the appropriate issue he can ask.

I am creating an ISSUE for each parameter  we think we might use and make a reference to this issue. 
All the issues/parameter can be worked in parallel.
Due to time constraints some of them are to be commented out for implementation in a future release.

This will give us a general skeleton. And if something at the end results redundant, not implemented, then we can just delete it or comment it out.
 
**Use boost program Options.**
http://www.boost.org/doc/libs/1_58_0/doc/html/program_options.html
some examples:
https://github.com/boostorg/program_options/tree/develop/example

I suggest that to any boost library that is going to be used, to gain experience:
1) make a “fast read” to the boost documentation, 
2) copy 2 or 3 small examples and execute them as they are
3) modify the examples a little, focusing on “how it could be integrated” into osm2pgrouting and  “see what happens”

**Comment by cvvergara** [comment]
For developing purposes in pgRouting we are using it here
https://github.com/pgRouting/pgrouting/blob/develop/src/ksp/src/mainP.cpp#L83

**Comment by sarthak-0415** [comment]
Read the documentation and coded the first phase. Code available on gsoc-develop branch

**Comment by cvvergara** [comment]
@dkastl @woodbri @sarthak-0415 
program_options shows the default, which do you think is best for the help message?
this:
```
  --prefixtables arg (=pgr_)            Add at the beginning of table names 
                                        (default: pgr_).
```
or:
```
  --prefixtables arg (=pgr_)            Add at the beginning of table names.
```

**Comment by sarthak-0415** [comment]
I think we should choose option 2 as showing the values twice is redundant

**Comment by cvvergara** [comment]
I agree with @sarthak-0415 opinion.

**Comment by woodbri** [comment]
I agree also.

**Comment by dkastl** [comment]
I agree as well.
But what about naming the prefix `osm_` by default. Since it's OSM data.

**Comment by cvvergara** [comment]
All the features requested in this discussion have being implemented in V2.1
